actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -253,11 +253,11 @@ def permit!
     #   ActionController::Parameters.new(person: {}).require(:person)
     #   # => ActionController::ParameterMissing: param is missing or the value is empty: person
     def require(key)
-      value = self[key]
-      if value.present? || value == false
-        value
+      if key.is_a?(Array)
+        keys.map &method(:require)
```

Comment:
Also lets create a method object, and just use the standard map invocation. That's 80% faster:

```ruby
require 'benchmark/ips'

def hello(key)
  key.upcase
end

keys = %i(a b c d e f)

Benchmark.ips do |x|
  x.report('method')   { keys.map(&method(:hello)) }
  x.report('standard') { keys.map { |k| hello(k) } }

  x.compare!
end
```

Results:

```
Calculating -------------------------------------
              method    22.731k i/100ms
            standard    37.384k i/100ms
-------------------------------------------------
              method    267.768k (± 2.4%) i/s -      1.341M
            standard    483.950k (± 2.9%) i/s -      2.430M

Comparison:
            standard:   483950.4 i/s
              method:   267768.2 i/s - 1.81x slower
```